tests/components/roborock/test_config_flow.py

```diff
@@ -178,3 +183,47 @@ async def test_config_flow_failures_code_login(
     assert result["data"] == MOCK_CONFIG
     assert result["result"]
     assert len(mock_setup.mock_calls) == 1
+
+
+async def test_reauth_flow(
+    hass: HomeAssistant, bypass_api_fixture, mock_roborock_entry: MockConfigEntry
+) -> None:
+    """Test reauth flow."""
+    # Fail authenticating on an already created entry.
+    with patch(
```

Comment:
This is great test coverage though, I believe the preference i usually see now is two split up the test responsibilities:
- tests in init that test exception handling and that reauth is triggered checking the state and `async_progress()` current step id
- tests in config flow that start a reauth flow via `async_init` (or maybe `async_start_reauth` is now the new way)